varname.py

```diff
@@ -417,6 +451,16 @@ def _node_name(node: ast.AST) -> str:
         return node.id
     if isinstance(node, ast.Attribute):
         return node.attr
+    if isinstance(node, (ast.List, ast.Tuple)):
+        if not all(isinstance(elem, (ast.Name, ast.Attribute))
+                   for elem in node.elts):
+            raise VarnameRetrievingError(
+                "All left-hand side should be either "
+                "a Name or an Attribute Node."
+            )
+        return tuple(elem.id if isinstance(elem, ast.Name)
+                     else elem.Attribute
+                     for elem in node.elts)
```

Comment:
```suggestion
        return tuple(_node_name(elem) for elem in node.elts)
```

This would mean the code can also handle something like `a, (b, c) = ...` and return the same structured tuple. varname users might find that useful, or they might prefer to just get a flat list of names or not allow this at all. What do you think? Otherwise you could still do something similar which only allows a flat tuple (like currently) but without duplicating logic.

Either way the current code is wrong, particularly `elem.Attribute`, which kind of demonstrates why reducing duplication would be good, and also that you probably need tests with multiple attributes on the LHS and a mix of attributes and names.

Reply:
Just Let it be able to handle the hierarchy and reusing the code is definitely a better idea. Thank you for the careful review. I really love it and appreciate it.